src/utils/svg.ts

```diff
@@ -124,3 +150,96 @@ export function initFonts(svg: Element) {
   }
   return false
 }
+
+/**
+ * Retrieves the attribute of an SVG and appends a suffix to ensure uniqueness.
+ * 
+ * @param attr String representing the name of the attribute to modify
+ * @param desc SVG to perform the modification on
+ * @param count Number representing the current value of the counter
+ */
+ export function replaceAttr(attr: string, desc: Element, count: number) {
+  var iri = desc.getAttribute(attr);
+  if (/^\s*#/.test(iri!)) { // Check if iri is non-null and internal reference
+    iri = iri?.trim()!;
+    desc.setAttribute(attr, iri! + suffix + count)
+  }
+}
+
+/**
+ * "Mangles" the SVG by appending a SVG-specific suffix to each element's ID to ensure
+ * each SVG component has an ID unique from other SVGs.
+ * 
+ * @param svg SVG element to "mangle"
+ */
+export function mangleSVG(svg: Element) {
+  // Variable declarations
+  let tagLabel = "";
+  let idRegex = /url\("?#([a-zA-Z][\w:.-]*)"?\)/g;
+  let iri_tag_map = new Map<string, number>();
+  let id_map = [];
+  let iriProperties = [];
+  var currentProp;
```

Comment:
```suggestion
  let currentProp;
```
var in javascript is global scope